pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+pynest microcicuit network
+----------------------------------------------
+
+Main file for the microcircuit.
+
+Hendrik Rothe, Hannah Bos, Sacha van Albada; May 2016
+'''
+
+import nest
+import numpy as np
+from helpers import *
+import matplotlib.pyplot as plt
+import os
+
+
+class Network:
+    """ Handles the setup of the network parameters and
+    provides functions to connect the network and devices.
+
+    Arguments:
+    sim_dict: dictionary containing all parameters specific to the simulation
+              such as the directory the data is stored in and the seeds
+              (see: stimulus_params.py)
+    net_dict: dictionary containing all parameters specific to the neurons
+              and the network
+              (see: network_params.py)
+
+    Keyword Arguments:
+    stim_dict: dictionary containing all parameter specific to the stimulus
+               (see: stimulus_params.py)
+    """
+    def __init__(self, sim_dict, net_dict, stim_dict=None):
+        self.sim_dict = sim_dict
+        self.data = sim_dict['data_path']
+        self.net_dict = net_dict
+        if stim_dict is not None:
+            self.stim_dict = stim_dict
+        print 'Data will be written to %s' % self.data
+
+    # hand parameters to NEST-kernel
+    def setup(self):
+        nest.ResetKernel()
+        master_seed = self.sim_dict['master_seed']
+        print 'master_seed ', master_seed
+        nest.SetKernelStatus({
+            'local_num_threads': self.sim_dict['local_num_threads']})
+        N_tp = nest.GetKernelStatus(['total_num_virtual_procs'])[0]
+        print 'number of total processes',  N_tp
+        rng_seeds = range(master_seed + 1 + N_tp, master_seed + 1 + (2 * N_tp))
+        grng_seed = master_seed + N_tp
+        print 'rng_seeds', rng_seeds
+        print 'grng_seed', grng_seed
+        self.pyrngs = [np.random.RandomState(s) for s in range(
+            master_seed, master_seed + N_tp)]
+        self.sim_resolution = self.sim_dict['sim_resolution']
+        kernel_dict = {
+                       'resolution': self.sim_resolution,
+                       'grng_seed': grng_seed,
+                       'rng_seeds': rng_seeds,
+                       'overwrite_files': self.sim_dict['overwrite_files'],
+                       }
+        nest.SetKernelStatus(kernel_dict)
+
+    # create neuronal populations
+    def create_populations(self):
+        self.N_ttl = self.net_dict['N_full']
```

Comment:
For clarity, I would recommend to name the parameters as in the dictionary, e.g., `self.N_full = self.net_dict['N_full']`.